### ISSUE TITLE
Error on creating many export rules in MetroCluster configuration

### ISSUE THREAD

**Issue by mamoep** [ansible-collections/netapp.ontap#47]
### Summary

I am creating a 30 export policy rules in a loop on a MetroCluster running ONTAP 9.8. I could replicate an issue where at rule 17 the play fails with the error:
`"msg": "Error on creating export policy Rule: calling: protocols/nfs/export-policies/64424509475/rules: got {u'message': u'This SVM is temporarily locked from changes while recording the configuration baseline. Try again in a few minutes or manually stop configuration replication by running \"vserver config-replication pause\" and retry the command.', u'code': u'2621573'}."`

All but one rules are created. Retrying the task creates the missing rules and succeeds.

### Component Name

na_ontap_export_policy_rule

### Ansible Version

```console
Tower 3.8.2 &
Ansible 2.9.19
```

### ONTAP Collection Version

```console
21.16
```

### ONTAP Version

```console
NetApp Release 9.8P7: Tue Sep 14 18:38:58 UTC 2021
```

### Playbook

```console
- name: Create export policy rules
    na_ontap_export_policy_rule:
      state: present
      name: foo
      vserver: svm-prod
      client_match: "{{ item.clientmatch }}"
      ro_rule: "{{ item.rorule }}"
      rw_rule: "{{ item.rwrule }}"
      protocol: nfs
      super_user_security: "{{ item.superuser }}"
      anonymous_user_id: "{{ item.anon | default(omit) }}"
      <<: *login
    loop: "{{ export_rules }}"
    loop_control:
      label: "HOST: {{ item.clientmatch }}; RO: {{ item.rorule }}; RW: {{ item.rwrule }}; ROOT: {{ item.superuser }};  ANON: {{ item.anon|default(65534) }}"
    when: export_rules_input | length > 0
    connection: local
```

### Steps to Reproduce

```yaml (paste below)
- hosts: localhost
  become: false
  collections:
    - netapp.ontap
  gather_facts: false
  vars:
    login: &login
      username: "DOMAIN\\user"
      password: foo
      hostname: cluster.local
      https: true
      validate_certs: true
    export_rules_input: |-
      foo01,sys,never,none
      foo10,sys,sys,sys
      foo11,sys,sys,sys
      foo12,sys,sys,sys
      foo13,sys,sys,sys
      foo14,sys,sys,sys
      foo15,sys,sys,sys
      foo16,sys,sys,sys
      foo17,sys,sys,sys
      foo18,sys,sys,sys
      foo19,sys,sys,sys
      bar01,sys,sys,sys
      bar02,sys,sys,sys
      bar03,sys,sys,sys
      bar04,sys,sys,sys
      bar05,sys,sys,sys
      bar06,sys,sys,sys
      foodb02,sys,sys,sys
      foodb01,sys,sys,sys
      bardb04,sys,never,none
      bardb03,sys,never,none
      bardb02,sys,never,none
      bardb01,sys,never,none
  
    export_rules_csv: "{{ export_rules_input.split('\n') }}"
  
  tasks:  
  - name: Convert CSV export rules to list of dict
    set_fact:
      export_rules: "{{ export_rules | default([]) + [ { \
                      'clientmatch': item.split(',').0,  \
                      'rorule':      item.split(',').1,  \
                      'rwrule':      item.split(',').2,  \
                      'superuser':   item.split(',').3,  \
                      'anon':        item.split(',').4 |default(65534) } ] }}"
    with_items: "{{ export_rules_csv }}"

  - name: Create export policy rules
    na_ontap_export_policy_rule:
      state: present
      name: foo
      vserver: svm-prod
      client_match: "{{ item.clientmatch }}"
      ro_rule: "{{ item.rorule }}"
      rw_rule: "{{ item.rwrule }}"
      protocol: nfs
      super_user_security: "{{ item.superuser }}"
      anonymous_user_id: "{{ item.anon | default(omit) }}"
      <<: *login
    loop: "{{ export_rules }}"
    loop_control:
      label: "HOST: {{ item.clientmatch }}; RO: {{ item.rorule }}; RW: {{ item.rwrule }}; ROOT: {{ item.superuser }};  ANON: {{ item.anon|default(65534) }}"
    when: export_rules_input | length > 0
    connection: local
```

### Expected Results

All rules should be created without error.

### Actual Results

```console
{
    "ansible_loop_var": "item",
    "_ansible_no_log": false,
    "invocation": {
        "module_args": {
            "username": "DOMAIN\\user",
            "key_filepath": null,
            "protocol": [
                "nfs"
            ],
            "ro_rule": [
                "sys"
            ],
            "feature_flags": {},
            "anonymous_user_id": "65534",
            "rule_index": null,
            "ontapi": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cert_filepath": null,
            "http_port": null,
            "name": "foo",
            "super_user_security": [
                "sys"
            ],
            "rw_rule": [
                "sys"
            ],
            "hostname": "cluster.local",
            "allow_suid": null,
            "vserver": "svm-prod",
            "state": "present",
            "https": true,
            "client_match": [
                "linux"
            ],
            "validate_certs": true,
            "use_rest": "auto"
        }
    },
    "item": {
        "rorule": "sys",
        "anon": 65534,
        "clientmatch": "linux",
        "rwrule": "sys",
        "superuser": "sys"
    },
    "changed": false,
    "msg": "Error on creating export policy Rule: calling: protocols/nfs/export-policies/64424509475/rules: got {u'message': u'This SVM is temporarily locked from changes while recording the configuration baseline. Try again in a few minutes or manually stop configuration replication by running \"vserver config-replication pause\" and retry the command.', u'code': u'2621573'}.",
    "_ansible_item_label": "HOST: linux; RO: sys; RW: sys; ROOT: sys;  ANON: 65534"
}
```

**Comment by carchi8py** [comment]
You may need to reach out to the ONTAP REST Api team on this. 

Looking at the code were calling the following API. 
https://library.netapp.com/ecmdocs/ECMLP2879871/html/#/NAS/export_rule_create
If it returns a job, we wait on the job to complete before returning, if it returns an error. we pass the error on to the user.

It looks like when the system temporarily locks itself to record the configuration baseline. It will just error out that the system is locked, instead of creating a job and putting in a queue for when the system is unlocked. 

You can send a message to the API team in the slack channel here https://netapp.io/ under the API channel.

**Comment by mamoep** [comment]
Could you handle this special case in the code and retry instead of error out?

**Comment by carchi8py** [comment]
I'll talk with the rest of the team and see if we can do something this and get back to you.

**Comment by mamoep** [comment]
Any news on this? I see this pop up in many places now all my playbooks are migrated to REST API. I workaround with ugly ansible retry mechanics.

**Comment by carchi8py** [comment]
Let me check with the team again.

**Comment by mamoep** [comment]
Fixed with release 22.6.0 

Thank you!
